kotlin/workflow-core/src/main/java/com/squareup/workflow/RenderContext.kt

```diff
@@ -62,12 +62,28 @@ interface RenderContext<StateT, in OutputT : Any> {
    *      }
    *    )
    *
+   * ## Equivalence & Testing
+   *
+   * It is common in unit tests to get a rendering from a workflow, then (since renderings should be
+   * value types) create the expected rendering instance and compare them. However, when comparing
+   * renderings there is no meaningful way to compare event handlers (other than a null check if
+   * they're nullable), and so two renderings that are identical but have different event handler
+   * functions should still be considered equal. In order to support this pattern, the functions
+   * returned by this method are all considered equal: `foo == bar` whenever `foo` and `bar` are
+   * values returned by this method. More precisely, this function returns instances of
+   * [EventHandler], and all [EventHandler] instances are considered equal.
```

Comment:
Is the last sentence still true?

Reply:
Yes. This doesn't change anything but the types. #385 requests changes to actual behavior, which will be done in a future PR.